src/openms/source/SYSTEM/File.cpp

```diff
@@ -140,6 +140,60 @@ namespace OpenMS
     return true;
   }
 
+  // https://stackoverflow.com/questions/2536524/copy-directory-using-qt
+  bool File::copyDirRecursively(const QString &fromDir, const QString &toDir, bool overwrite_existing)
+  {
+  
+    QDir sourceDir(fromDir);
+    QDir targetDir(toDir);
+
+    QString canonical_sourceDir = sourceDir.canonicalPath();
+    QString canonical_targetDir = targetDir.canonicalPath();
+   
+    // check canonical path  
+    if (canonical_sourceDir == canonical_targetDir)
+    {
+      LOG_ERROR << "Error: Could not copy  " << fromDir.toStdString() << " to " << toDir.toStdString() << ". Same path given." << std::endl;;  
+      return false;
+    }
+
+
+    // make directory if not present 
+    if (!targetDir.exists())
+    {
+      targetDir.mkpath(toDir);
+    }
+  
+    // copy folder recurively
+    QFileInfoList fileInfoList = sourceDir.entryInfoList();
+    foreach(QFileInfo fileInfo, fileInfoList)
```

Comment:
does this also work with standard C++11 now?, i.e. `for (QFileinfo& file_info : fileinfo_list)` ??

Reply:
@cbielow thank you for your patience - will work on it.